src/nodes.coffee

```diff
@@ -799,20 +799,27 @@ exports.Block = class Block extends Base
       'BlockStatement'
 
   astProperties: (o) ->
+    checkForDirectives = del o, 'checkForDirectives'
+
+    sniffDirectives @expressions, notFinalExpression: checkForDirectives if @isRootBlock or checkForDirectives
```

Comment:
`sniffDirectives()` replaces expressions that should be treated as directives with `Directive` node class instances